prql-compiler/tests/integration/main.rs

```diff
@@ -30,258 +48,136 @@ mod tests {
                 return;
             }
 
-            // compile
-            let opts = Options::default()
-                .with_target(Target::Sql(Some(Dialect::SQLite)))
-                .no_format();
-            let sql = prql_compiler::compile(&prql, &opts)
-                .unwrap_or_else(|e| panic!("Failed to compile\n\n{prql}\n\n{e}"));
-
-            let sqlite_out = sqlite::query_csv(&sqlite_conn, &sql);
+            let mut results = BTreeMap::new();
+            for con in &mut connections {
+                let vendor = con.get_dialect().to_string().to_lowercase();
+                if prql.contains(format!("skip_{}", vendor).as_str()) {
+                    continue;
+                }
+                results.insert(vendor, run_query(con.as_mut(), prql.as_str(), &runtime));
+            }
 
-            // save both csv files as same snapshot
-            let opts = Options::default()
-                .with_target(Target::Sql(Some(Dialect::DuckDb)))
-                .no_format();
-            let sql = prql_compiler::compile(&prql, &opts).unwrap();
-            let duckdb_out = duckdb::query_csv(&duckdb_conn, &sql);
-            pretty_assertions::assert_eq!(sqlite_out, duckdb_out, "SQLite == DuckDB: {test_name}");
+            if results.is_empty() {
+                return;
+            }
 
-            if let Some(pg_client) = &mut pg_client {
-                let opts = Options::default()
-                    .with_target(Target::Sql(Some(Dialect::PostgreSql)))
-                    .no_format();
-                let sql = prql_compiler::compile(&prql, &opts).unwrap();
-                let pg_out = postgres::query_csv(pg_client, &sql);
-                pretty_assertions::assert_eq!(sqlite_out, pg_out, "SQLite == PG: {test_name}");
+            let first_result = match results.iter().next() {
+                Some(v) => v,
+                None => return,
+            };
+            for (k, v) in results.iter().skip(1) {
+                pretty_assertions::assert_eq!(
+                    *first_result.1,
+                    *v,
+                    "{} == {}: {test_name}",
+                    first_result.0,
+                    k
+                );
             }
```

Comment:
Your implementation is probably better at debugging, so feel completely free to keep it.

That said, I think it's possible to simplify it a bit — here's one approach. The result is basically the same (the only difference is that the snapshot file contains the input prql as the expression, because of the third item passed to `assert_snapshot!`)

```rust

    #[test]
    fn test_rdbms() {
        let runtime = Runtime::new().unwrap();
        let mut connections = get_connections(&runtime);

        for con in &mut connections {
            setup_connection(con.as_mut(), &runtime);
        }

        // for each of the queries
        glob!("queries/**/*.prql", |path| {
            // read
            let prql = fs::read_to_string(path).unwrap();

            if prql.contains("skip_test") {
                return;
            }

            for con in &mut connections {
                let vendor = con.get_dialect().to_string().to_lowercase();
                if prql.contains(format!("skip_{}", vendor).as_str()) {
                    continue;
                }
                let result = run_query(con.as_mut(), prql.as_str(), &runtime);
                let result_string = result.iter().map(|row| row.join(",")).join("\n");

                assert_snapshot!("rdbms".to_string(), &result_string, &prql);
            }
        });
    }
```

(hopefully this is helpful feedback, rather than dispiriting that someone is "rewriting your code", let me know either way)